packages/auth-remix/src/server.ts

```diff
@@ -0,0 +1,795 @@
+    req: Request,
+    cb: (params: ParamsOrError<{ tokenData: TokenData }>) => Res | Promise<Res>
+  ): Promise<Res>;
+  async emailPasswordSendPasswordResetEmail<Res extends Response>(
+    req: Request,
+    data: { email: string; password: string },
+    cb: (params: ParamsOrError<{ tokenData: TokenData }>) => Res | Promise<Res>
+  ): Promise<Res>;
+  async emailPasswordSendPasswordResetEmail(
+    req: Request,
+    dataOrCb?:
+      | { email: string; password: string }
+      | ((
+          params: ParamsOrError<{ tokenData: TokenData }>
+        ) => Response | Promise<Response>),
+    cb?: (
+      params: ParamsOrError<{ tokenData: TokenData }>
+    ) => Response | Promise<Response>
+  ): Promise<
+    | Response
+    | {
+        tokenData: TokenData;
+        headers: Headers;
+      }
+  > {
+    return handleAction(
+      async (data, headers) => {
+        if (!this.options.passwordResetPath) {
+          throw new Error(`'passwordResetPath' option not configured`);
+        }
+        const [email] = _extractParams(data, ["email"], "email missing");
+
+        const { verifier } = await (
+          await this.core
+        ).sendPasswordResetEmail(
+          email,
+          new URL(
+            this.options.passwordResetPath,
+            this.options.baseUrl
+          ).toString()
+        );
+
+        headers.append(
+          "Set-Cookie",
+          `${this.options.pkceVerifierCookieName}=${verifier}; HttpOnly; SameSite=strict`
+        );
+      },
+      req,
+      dataOrCb,
+      cb
+    );
+  }
+
+  async emailPasswordResetPassword(
+    req: Request,
+    data?: { email: string; password: string }
+  ): Promise<{ tokenData: TokenData; headers: Headers }>;
+  async emailPasswordResetPassword<Res extends Response>(
+    req: Request,
+    cb: (params: ParamsOrError<{ tokenData: TokenData }>) => Res | Promise<Res>
+  ): Promise<Res>;
+  async emailPasswordResetPassword<Res extends Response>(
+    req: Request,
+    data: { email: string; password: string },
+    cb: (params: ParamsOrError<{ tokenData: TokenData }>) => Res | Promise<Res>
+  ): Promise<Res>;
+  async emailPasswordResetPassword(
+    req: Request,
+    dataOrCb?:
+      | { email: string; password: string }
+      | ((
+          params: ParamsOrError<{ tokenData: TokenData }>
+        ) => Response | Promise<Response>),
+    cb?: (
+      params: ParamsOrError<{ tokenData: TokenData }>
+    ) => Response | Promise<Response>
+  ): Promise<
+    | Response
+    | {
+        tokenData: TokenData;
+        headers: Headers;
+      }
+  > {
+    return handleAction(
+      async (data, headers, req) => {
+        const verifier = parseCookies(req)[this.options.pkceVerifierCookieName];
+
+        if (!verifier) {
+          throw new Error("no pkce verifier cookie found");
+        }
+
+        let resetToken = new URL(req.url).searchParams.get("reset_token");
+
+        if (Array.isArray(resetToken)) {
+          resetToken = resetToken[0];
+        }
```

Comment:
Don't think this is needed, `searchParams.get` always returns `string | null`. It's only an array if you use `searchParams.getAll`

Reply:
Also why is the `reset_token` in the searchParams and not the body?

Reply:
I updated this to use reset_token from the data